relational_operators/DropTableOperator.hpp

```diff
@@ -48,9 +48,9 @@ class DropTableOperator : public RelationalOperator {
    * @param only_drop_blocks If true, only drop the blocks belonging to
    *        relation, but leave relation in the database.
    **/
-  explicit DropTableOperator(const relation_id rel_id,
+  explicit DropTableOperator(const CatalogRelation &relation,
```

Comment:
A quick answer is that frontend can't generate the relation_id - that happens in the backend.

To give more details, here I explain a bit more about the big picture. In our plan, the frontend will receive a readonly copy of the catalog, then performs query planning / optimization and generates the query plan. The query plan will be executed by the backend.

So, the frontend's output will contain four messages that are encoded into protobuf:
(1) The query context.
(2) The DAG of relational operators.
(3) The temporary catalog relations.
(4) A reference to the result relation.

The backend, upon receiving the query plan, will do the follows:
(1) First, reconstruct query context, DAG of relational operators, temporary relations and the reference to the result relation.
(2) Insert temporary relations into the catalog database.
(3) Execute the DAG of relational operators. Note that temporary relations are dropped during execution.
(4) Return results, then drop the (temp) result relation.

About why relation_id is not sufficient. Here is one example.

Suppose currently the catalog database has two stored relations R and S, which have relation_id 0 and 1. Now we are to execute a complex query Q1 so three temporary relations A, B, C are generated. The question is, what is the relation_id of A, B, C? At first glance, we might use 2, 3, 4. But it is actually *resolving relation_id at the frontend*, which endures the loss of isolation.

That is, suppose just after Q1 is planned but its query plan has not yet arrived at the backend, some query Q2 is also planned that occupies catalog database slot 2, 3 and gets executed first. Then when Q1 arrives at the backend, it will see a confliction, and it becomes nasty to modify the relation_id's at this time point - you have to dive into every corner of query context and relational operators to inject code to adjust the relation_id's.

One thing is true is that from the pure backend perspective, relation_id is sufficient. But currently the data structures (especially relational operators) are shared by the both the frontend and backend. So use CatalogRelation& to replace relation_id is the most convenient way. Note that the pure backend part -- WorkOrders of relational operators -- are not affected. They'e still using relation_id's.